Automate code linting and reject PR if fails.

mmphego opened the pull request:
This commit does the following:

- Added default setup.cfg (Refer to [CAM Developer's Guide](https://docs.google.com/document/d/1aZoIyR9tz5rCWr2qJKuMTmKp2IzHlFjrCFrpDDHFypM/edit#heading=h.whqqrvn0pw8f) which adds PEP8 strictness.)
- Added pyproject.toml config file which contains isort and black configurations/parameters.
- Added linting to static analysis stage on Jenkins

JIRA: [MT-1063](https://skaafrica.atlassian.net/browse/MT-1063)

mmphego committed:
Automate code linting and reject PR if fails.

This commit does the following:

- Added default setup.cfg (Referencing [CAM
guidelines](https://docs.google.com/document/d/18lHeyVoPtOGrQzbLlT-2cacDRzMn_Oxvg3unhuyPK3o/edit))
- Added pyproject.toml config file which contains isort and black configurations/parameters.
- Added linting as part of travis ci, before script execution

JIRA: [MT-1063](https://skaafrica.atlassian.net/browse/MT-1063)

mmphego committed:
Updated travis.yml with correct env config

mmphego committed:
Try to fetch refs from github ssh, if fails fetch from https

ajoubertza commented on lint_diff.sh:
```diff
@@ -0,0 +1,146 @@
+# License: BSD 3 clause
+# Ref: https://github.com/sardana-org/sardana/blob/develop/ci/lint_diff.sh
+
+# This script is used on Jenkins CI (linting stage) to check that PRs does not add
```
This package happens to use Travis, not Jenkins.

ajoubertza reviewed (approved):
Nice.  Good idea to include the script in the repo.  One tiny change.

mmphego committed:
Updated script description.
Removed pvt cam modules from isort config.
Removed non documented flake8 violation code.

bngcebetsha reviewed (approved):
LGTM
